backend/zane_api/views/auth.py

```diff
@@ -168,3 +173,53 @@ def get(self, _: Request) -> Response:
 
         if response.is_valid():
             return Response(response.data)
+
+
+class UserExistenceResponseSerializer(serializers.Serializer):
+    exists = serializers.BooleanField()
+
+
+class UserCreationSerializer(serializers.Serializer):
+    username = serializers.CharField(min_length=1, max_length=255)
+    password = serializers.CharField(min_length=6, write_only=True)
+
+
+class CheckUserExistenceView(APIView):
+    permission_classes = [permissions.AllowAny]
+
+    @extend_schema(
+        responses={200: UserExistenceResponseSerializer},
+        summary="Check if a user exists",
+        description="Returns whether a single user already exists in the system.",
+    )
+    def get(self, request):
+        exists = User.objects.count() == 1
+        return Response({"exists": exists}, status=status.HTTP_200_OK)
+
+
+class CreateUserView(APIView):
+    permission_classes = [permissions.AllowAny]
+
+    @extend_schema(
+        request=UserCreationSerializer,
+        responses={201: None, 400: None},
+        summary="Create a user",
+        description="Creates a new user if no user exists.",
+    )
+    def post(self, request):
```

Comment:
```suggestion
    def post(self, request: Request) -> Response:
```

We use typing in python, see the other endpoints in the file 🙂 .